### ISSUE TITLE
Support MP3 audio on native targets

### ISSUE THREAD

**Issue by joshtynjala** [openfl/openfl#2441]
**Is your feature request related to a problem? Please describe.**
-----

I would like to be able to play .mp3 files on native targets. As I understand it, this was not originally supported in OpenFL/Lime because the MP3 format was under patent at the time, and there were legal concerns about including a native MP3 playback library. Now, I believe that the MP3 patents are expired, and it should not cause legal issues to include a library.

**Describe the solution you'd like**
-----

MP3 support on native targets should use the standard `Sound` API, like how it already works on other targets where MP3 support is provided by the target.

**Describe alternatives you've considered**
-----

The best alternative would be to convert the .mp3 files to another format. Unfortunately, my project involves using MP3s supplied by users. It is not feasible to ask my users to do the conversion manually, or to make them wait the extra time my app would need to run an encoder automatically.

**Comment by dimensionscape** [comment]
Yes, yes, yes! Thanks for bringing this up!

**Comment by dimensionscape** [comment]
https://sourceforge.net/projects/lame/files/lame/3.100/
https://en.wikipedia.org/wiki/LAME#Patents_and_legal_issues

Edit*** Oh its just an encoder, but hey, theres hope.

**Comment by joshtynjala** [comment]
Related Lime issue: haxelime/lime#892

**Comment by loudoweb** [comment]
source (wikipedia):
> The basic MP3 decoding and encoding technology is patent-free in the European Union, all patents having expired there by 2012 at the latest. In the United States, the technology became substantially patent-free on 16 April 2017

**Comment by flustix** [comment]
Has there been any fix/implementation for this yet?
Because I would really need it for a game I'm working on.

**Comment by joshtynjala** [comment]
@Flustix No, no one has worked on this yet. I investigated a bit myself, but I'm not very good at C/C++, so I got stuck.

**Comment by dimensionscape** [comment]
Actually, I have built a reasonably fast codec purely in Haxe for decoding. It's currently in early/experimental stages and only layer III is supported for now. Let me know if you are interested in contributing code towards this improvement or sponsoring financially. 

Thanks for your interest!

> Has there been any fix/implementation for this yet? Because I would really need it for a game I'm working on.

**Comment by flustix** [comment]
I'd like to contribute but i have no idea how audio encoding works.
Still looking forward to it because i dont want my users to have to convert the audio files.
